frontend/src/views/HomeView.vue

```diff
@@ -1 +1,89 @@
-<template><h1>Test</h1></template>
+<script lang="ts" setup>
+import { onMounted, ref, reactive } from 'vue'
+import { ElButton, ElDialog, ElForm, ElFormItem, ElInput, ElMessage } from 'element-plus'
+import { useRolesStore } from '@/stores/roles/index'
+import DemoSection from './DemoSection.vue'
+
+const dialogFormVisible = ref(false)
+const formLabelWidth = '100px'
+
+const rolesStore = useRolesStore()
+const newRoleName = ref('')
+
+enum Views {
+  references = 'references',
+  demo = 'demo'
+}
+const activeView = ref(Views.references)
+
+onMounted(async () => {
+  await rolesStore.fetchRoles()
+})
+
+const createRole = async () => {
+  if (newRoleName.value == '') {
+    ElMessage({
+      message: 'you need to fill in this field',
+      type: 'error'
+    })
+  } else {
+    dialogFormVisible.value = false
+    try {
+      await rolesStore.addRole({ name: newRoleName.value })
+
+      ElMessage({
+        message: 'Congrats, your role is now added',
+        type: 'success'
+      })
+    } catch (error: Error) {
+      if (error.status === 409) {
+        ElMessage({
+          message: 'Role name already exists.',
+          type: 'error'
+        })
+      } else {
+        // Handle other status codes
+        ElMessage({
+          message: 'unexpected error',
+          type: 'error'
+        })
+      }
+    }
+  }
+
+  newRoleName.value = ''
+}
+</script>
+
+<template>
+  <div class="demo">
+    <ElButton plain @click="dialogFormVisible = true"> Add new Role </ElButton>
+    <ElDialog v-model="dialogFormVisible" title="New Role" width="500">
+      <ElInput v-model="newRoleName" placeholder="Role name" :clearable="false" />
+
+      <template #footer>
+        <div class="dialog-footer">
+          <ElButton @click="dialogFormVisible = false">Cancel</ElButton>
```

Comment:
Same here. Please, avoid to put logic inside template tag, always inside script tag